### PR TITLE
[Debugger] Improve tooltip sizing on catalina

### PR DESCRIPTION
**Pull request by sgmunn**
Fixes VSTS 1002713

This still seems to have issues where the tooltip does not reduce in size when collapsing rows though.

### CODE REVIEW

**Comment by jstedfast** [comment]
This regresses the tooltips on Mojave. If I expand a bunch of nodes and then collapse them, the tooltip width doesn't shrink.

Not only that, but when I collapse, the value ends up way out in Boonesville:

boonesville

**Comment by sgmunn** [comment]
for the record, same behavior on catalina. I was trying to at least get the tooltips to resize on catalina.

:(

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.VSTextView/QuickInfo/MacDebuggerTooltipWindow.cs [hunk 1]
```diff
@@ -108,12 +108,14 @@ static nfloat GetMaxHeight (NSWindow window)
 
 		void OnTreeViewResized (object sender, EventArgs e)
 		{
+			Console.WriteLine ("OnTreeViewResized: treeView.Frame.Width = {0}", treeView.Frame.Width);
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.VSTextView/QuickInfo/MacDebuggerTooltipWindow.cs]: Debug code.

Comment (jstedfast) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.VSTextView/QuickInfo/MacDebuggerTooltipWindow.cs]: yea, this entire PR is a Work-in-Progress.

Debugger Tooltips on Catalina went from 100% broken to 50% broken, but this PR also makes Debugger Tooltips on Mojave go from 0% broken to 50% broken.

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs [hunk 2]
```diff
@@ -158,9 +158,16 @@ protected static NSAttributedString GetAttributedPlaceholderString (string text)
...
-		protected nfloat GetWidthForString (NSControl control, string text)
+		protected static nfloat GetWidthForString (NSFont font, string text, int sizeDelta = 0)
 		{
-			var str = new NSAttributedString (text, font: control.Font);
+			NSFont modified = null;
+
+			if (sizeDelta != 0)
+				modified = NSFont.FromDescription (font.FontDescriptor, font.PointSize + sizeDelta);
+
+			var str = new NSAttributedString (text, font: modified ?? font);
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs]: Nit: `using var str`

**Comment by sgmunn** [comment]
```
System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object) ---> System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Debugger.MacDebuggerObjectCellViewBase.GetWidthForString (AppKit.NSFont font, System.String text, System.Int32 sizeDelta) [0x00005] in /Users/runner/runners/2.160.0/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs:167 
  at MonoDevelop.Debugger.MacDebuggerObjectValueView.GetOptimalWidth (MonoDevelop.Debugger.MacObjectValueTreeView treeView, MonoDevelop.Debugger.ObjectValueNode node, System.Boolean hideValueButton) [0x00209] in /Users/runner/runners/2.160.0/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectValueView.cs:393 
  at MonoDevelop.Debugger.MacObjectValueNode.Measure (MonoDevelop.Debugger.MacObjectValueTreeView treeView) [0x000b3] in /Users/runner/runners/2.160.0/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueNode.cs:81 
  at MonoDevelop.Debugger.MacObjectValueTreeView.OptimizeColumnSizes (System.Boolean emitResized) [0x0004b] in /Users/runner/runners/2.160.0/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeView.cs:228 
  at MonoDevelop.Debugger.MacObjectValueTreeView.ExpandItem (Foundation.NSObject item, System.Boolean expandChildren) [0x00025] in /Users/runner/runners/2.160.0/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeView.cs:356 
```

**Comment by sgmunn** [comment]
We can abandon this. work on this was taken up here - https://github.com/mono/monodevelop/pull/9270

**Pull request closed without merging**